enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,854 @@
+
+`openshift-tests` will plan an efficient testing strategy based on the 
+extension binaries it discovers, the configuration profiles they support,
+and the tests that need to run in each configuration. It will:
+1. Discover extension binaries.
+2. Discover extension binary metadata (`info` verb), including configuration profiles.
+3. Discover tests that should be run for each configuration profile (`list` verb with `--config <profile>`, for each profile).
+4. Plan test execution by grouping configuration profiles into buckets.
+
+After `openshift-tests` plans test orchestration, it will proceed to execute tests in
+each configuration bucket. For a given bucket, `openshift-tests`
+will:
+1. Iterate through all involved components and configure them with non-conflicting configuration profiles (`config` verb).
+2. Await a high-level cluster steady state (cluster operators available, machines config up-to-date) if any configuration profile requires it.
+3. Run tests within the bucket with non-conflicting, test-level parallelization.
+
+Between buckets, the component may be asked to set its required `default` profile in order
+to return to its install-time configuration.
+
+#### Update - Metadata Validation
+Component owners will be responsible for implementing the extension interface. To prevent common mistakes
+and ensure conformance with the evolution of their implementation, `make` (or similar build system)
+must run `<extension binary> update [--component product:type:component] [--basedir <basedir>]` after the extension binary is built.
+
+The `update` verb will create or update files under `hack/.openshift-tests-exension/product/type/component/*`, by default
+(basedir defaults to `hack/.openshift-tests-extension`).
+If an incompatible change is introduced from the prior invocation of `update` (e.g. changing 
+a test name without preserving the original), `update` will raise an error 
+which the component owner must correct before committing their change in git.
+
+The content of the files stored under `.openshift-tests-extension` is subject to change and wholly
+defined by the `github.com/openshift-eng/openshift-tests-extension` module. Individual component
+owners should not modify files under this path except through invocations of `update`.
+
+#### Extension Implementation
+github.com/openshift-eng/openshift-tests-extension will make contributing tests as simple
+as possible. An example of what implementing the extension interface with this module
+is provided below.
+
+```golang
+componentExtension := extensions.NewExtension("openshift", "payload", "hyperkube")
+
+// boilerplate to initialize ginkgo omitted
+
+// Allow extensions to Walk tests and return them wrapped
+// in a wrapper extensions.ExtensionTestSpec. We can offer these
+// helper functions to ingest tests from different testing framework.
+var ginkgoExtensionSpecs extensions.ExtensionTestSpecSet  // an abstraction around a group of ExtensionTestSpec

```

Comment:
I used ExtensionTestSpecs, which is consistent with ginkgo's plural collection of objects (e..g Label/Labels)